### ISSUE TITLE
Tasks tab in web UI nukes GPU performance

### ISSUE THREAD

**Issue by soxrok2212** [hashtopolis/server#722]
- Server Version: 0.12.0 commit 2be8530
- Client Version: 0.6.1.3
- Hashcat v6.2.3
- Task doesn't matter, but in this case is -a 3 ?d?d?d?d?d?d?d?d?d?d on mode 22000 (WPA)
- Debug from client probably not helpful here

Issue: It's broke! jk, This issue has persisted for at least a year but it was hugely exaggerated on an RTX 3080 I just picked up. This 3080 is in my personal desktop, which is also where I manage my tasks from the web GUI. When a task is running on the GPU, if you go into the Show Tasks tab, the hashcat speed will drop (in this case on the 3080, it was SIGNIFICANT, down from ~910kH/s to 500-700kH/s).

It was also apparent on my 1080, though not as dramatic. The issue persists if I go into a specific task as well. Not temps related, I'm staying at or under 70c and when I close the tab, or say go to the agents tab, speeds kick back up.

OS is Win 10 64 bit latest updates, browser is Firefox, also latest updates. Looks like some of the artifacts on the show tasks pages are hogging resources really bad. Latest Nvidia drivers, using CUDA as well.

Not sure how else I can diagnose this, but can certainly provide more info if you tell me how/what you need.

**Comment by soxrok2212** [comment]
![firefox](https://user-images.githubusercontent.com/6226507/135690625-bda78823-db1a-484c-a969-7072146e1d58.PNG)
Quick SS, right at the drop is where i close Firefox and speeds resume to normal.

Chome does not seem to exhibit this behavior as bad, though it does seem to have small "blips" of higher gpu usage, particularly when auto-refresh is on. Nothing appears sustained as it does in Firefox.
![chrome](https://user-images.githubusercontent.com/6226507/135690767-7b54c218-d1fa-46ed-924f-b0988467e36d.PNG)

**Comment by soxrok2212** [comment]
On a final note, disabling hardware acceleration significantly improves this issue on the Firefox side.

**Comment by zyronix** [comment]
Based on what you are describing, you are running Firefox on the same node you are also cracking with? Firefox uses the GPU for rendering pages, if a page with a bit more information has to be render this might cause your cracking job the slowdown. It would be the same if you were gaming for example on your computer while you are also cracking password.

A solution could indeed be to disable hardware acceleration or add a dedicated GPU for cracking and one for rendering your screen/apps.

**Comment by soxrok2212** [comment]
Yes I did that and it significantly improved the performance. I prefer headless nodes and most are except for this one. 

I’ll leave it up to @s3inlc to close or not as we have a “solution”, though I do believe it is still not optimal. Where the issue lies (Firefox or Hashtopolis) I don’t know for sure, as it’s not nearly as drastic in chrome.

**Comment by s3inlc** [comment]
I agree with @zyronix that this is probably not something we can really tackle. And with the hardware acceleration disabled it should be acceptable as a solution. I just assume that Chrome is either having some rendering implemented in a more efficient way or not using that much hardware acceleration as Firefox.
